toolkit/components/uniffi-js/UniFFIPointerType.cpp

```diff
@@ -0,0 +1,20 @@
+/* -*- Mode: C++; tab-width: 2; indent-tabs-mode: nil; c-basic-offset: 2 -*- */
+/* vim:set ts=2 sw=2 sts=2 et cindent: */
+/* This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this
+ * file, You can obtain one at http://mozilla.org/MPL/2.0/. */
+
+#include "mozilla/dom/UniFFIPointerType.h"
+
+namespace mozilla::dom {
+
+void UniFFIPointerType::getTypeName(nsString& retval) {
+  retval = this->typeName;
+}
+
+void UniFFIPointerType::destroyPtr(void *ptr) {
+  RustCallStatus status;
+  this->destructor(ptr, &status);
```

Comment:
TODO: do something with the status here if the destructor call fails... not sure what to do though, but at least some type of log? the destructor failing means something is fundementally wrong..

Reply:
maybe an NS_ASSERT would be enough, at least initially?